File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int l=0;
        int h=nums.length-1;
        int res=-1;
        while(l<h){
            int guess=l+(h-l)/2;
            if(nums[guess]>nums[h]){
                   l=guess+1;
            }else{
               h=guess;
            }
        }
        return nums[l];
        
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left=0;
        int right=nums.size()-1;
        while(left<right){
            int m=left+(right-left)/2;
            if(nums[m]>nums[right])
            left=m+1;
            else
            right=m;
        }
        return nums[left];
        
    }
};